src/dtls.c

```diff
@@ -1372,10 +1406,40 @@ int wolfSSL_dtls_cid_get_tx(WOLFSSL* ssl, unsigned char* buf,
     return DtlsCidGet(ssl, buf, bufferSz, 0);
 }
 
+int wolfSSL_dtls_cid_get0_tx(WOLFSSL* ssl, unsigned char** cid)
+{
+    return DtlsCidGet0(ssl, cid, 0);
+}
+
 int wolfSSL_dtls_cid_max_size(void)
 {
     return DTLS_CID_MAX_SIZE;
 }
+
+void wolfSSL_dtls_cid_parse(const unsigned char* msg, unsigned int msgSz,
```

Comment:
`cid` is already the return. I don't see the point of complicating with an extra return. If anything, I can have the function return the pointer instead of outputting it to a parameter.

Reply:
ok, I do think checking for the return is more uniform than checking if *cid is NULL. Please add this to the documentation of the API.

Reply:
Ok so I'll change it to return the CID instead of setting the parameter.